### PR TITLE
feat(enrichment_tables): add support for custom MMDB types

### PR DESCRIPTION
**Pull request by esensar**
This adds support for custom MMDB types. It will just return whatever is stored in the database, without further modifications. Test data was generated using the official go example:
https://github.com/maxmind/mmdbwriter/blob/main/examples/asn-writer/main.go

Fixes: #19995

### CODE REVIEW

**Comment by esensar** [comment]
Does this make sense in the context of #19995, or did I misunderstand the issue?

**Review by jszwedko — COMMENTED** [review]
Nice! This was easier than I expected. Could you add a changelog entry? I'm also wondering if this could be considered a "breaking" change? It _seems_ like no since any city databases will have `GeoIP-City` as the type. The only thing it would break is if someone had a custom type that happened to match the `GeoIP-City` format. That seems like enough of an edge case that we could consider it non-breaking even though it technically is.

**Comment by esensar** [comment]
> Nice! This was easier than I expected. Could you add a changelog entry? I'm also wondering if this could be considered a "breaking" change? It _seems_ like no since any city databases will have `GeoIP-City` as the type. The only thing it would break is if someone had a custom type that happened to match the `GeoIP-City` format. That seems like enough of an edge case that we could consider it non-breaking even though it technically is.

I guess it can break stuff, but since that would be kind of an undocumented usage, maybe it is not that bad?

**Review by jszwedko — COMMENTED** [review]
Per https://github.com/vectordotdev/vector/issues/19995#issuecomment-1992242523 I think we want to model the custom mmdb reading as a completely new enrichment table type (`mmdb`).

File: website/cue/reference/configuration.cue [hunk 1]
```diff
@@ -220,6 +220,9 @@ configuration: {
 						* [GeoIP2-ISP.mmdb](\(urls.maxmind_geoip2_isp)) (paid) — Determine the Internet
 							Service Provider (ISP), organization name, and autonomous system organization
 							and number associated with an IP address.
+						* Custom - any other database in MaxMind DB file format, which does not
+							match above mentioned types. Fields are returned as they are stored in
```
Comment (urseberry) [inline website/cue/reference/configuration.cue:224]:
```suggestion
							match the above mentioned types. Fields are returned as they are stored in
```

File: src/enrichment_tables/geoip.rs [hunk 2]
```diff
@@ -18,8 +18,8 @@ use vrl::value::{ObjectMap, Value};
 use crate::config::{EnrichmentTableConfig, GenerateConfig};
 
 // MaxMind GeoIP database files have a type field we can use to recognize specific
-// products. If we encounter one of these two types, we look for ASN/ISP information;
-// otherwise we expect to be working with a City database.
+// products. If it is an unknown type, it will be treated as a Custom kind and will
+// directly read data from the database.
```
[the remark this answers is not in the document]
Comment (esensar) [reply src/enrichment_tables/geoip.rs:22]: Oops, missed that. You are right, it is no longer accurate.

**Comment by jszwedko** [comment]
Thanks @esensar !

**Pull request merged**